src/overlays/ovl_DF4260/fox_co.c

```diff
@@ -0,0 +1,3333 @@
+        *arg1 = NULL;
+    }
+    return 0;
+}
+
+void func_DF4260_8018EE84(Boss* boss) {
+
+    Matrix_Translate(gGfxMatrix, -D_DF4260_80199A90.x, -D_DF4260_80199A90.y, 0.0f, 1);
+    Matrix_SetGfxMtx(&gMasterDisp);
+    Animation_GetFrameData(D_602D400, 0, boss->vwork);
+    Animation_DrawSkeleton(1, D_602D5AC, boss->vwork, func_DF4260_8018EE2C, NULL, &boss->index, &gIdentityMatrix);
+}
+
+s32 func_DF4260_8018EF38(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    if (arg0 == 3) {
+        arg3->z -= gBosses[*(s32*) arg4].fwork[1];
+    }
+    if ((arg0 != 3) && (arg0 != 4)) {
+        *arg1 = NULL;
+    }
+    return 0;
+}
+
+void func_DF4260_8018EF90(Boss* boss) {
+
+    Matrix_Translate(gGfxMatrix, -D_DF4260_80199AA8.x, -D_DF4260_80199AA8.y, 0.0f, 1);
+    Matrix_SetGfxMtx(&gMasterDisp);
+    Animation_GetFrameData(D_602D400, 0, boss->vwork);
+    Animation_DrawSkeleton(1, D_602D5AC, boss->vwork, func_DF4260_8018EF38, NULL, &boss->index, &gIdentityMatrix);
+}
+
+extern UNK_TYPE D_603E924;
+
+void func_DF4260_8018F044(Object_80* obj80) {
+
+    switch (obj80->unk_48) {
+        case 0:
+            if (obj80->unk_50 != 0) {
+                obj80->unk_50 = 0;
+                if (obj80->unk_54 < 2) {
+                    obj80->unk_44++;
+                    obj80->timer_4C = 5;
+                    obj80->vel.x += 2.0f;
+                    obj80->vel.y += 2.0f;
+                    if (obj80->unk_44 >= 3) {
+                        obj80->unk_48 = 1;
+                        obj80->info.hitbox = SEGMENTED_TO_VIRTUAL(&D_603E924);
+                        Audio_PlaySfx(0x1900000D, &obj80->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                        return;
+                    }
+                    Audio_PlaySfx(0x2903300E, &obj80->sfxPos, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                }
+            }
+            break;
+        case 1:
+            Math_SmoothStepToF(&obj80->vel.x, 80.0f, 0.2f, 10.0f, 0.0f);
+            Math_SmoothStepToF(&obj80->vel.y, 80.0f, 0.2f, 10.0f, 0.0f);
+            break;
+    }
+}
+
+s32 func_DF4260_8018F1C8(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    Object_80* obj80 = (Object_80*) arg4;
+
+    RCP_SetupDL_29(gFogRed, gFogGreen, gFogBlue, gFogAlpha, gFogNear, gFogFar);
+    switch (arg0) {
+        case 1:
+            arg3->y -= obj80->vel.x;
+            if (obj80->timer_4C & 1) {
+                RCP_SetupDL_60(gFogRed, gFogGreen, gFogBlue, gFogAlpha, gFogNear, gFogFar);
+            }
+            break;
+        case 2:
+            arg3->y += obj80->vel.y;
+            if (obj80->timer_4C & 1) {
+                RCP_SetupDL_60(gFogRed, gFogGreen, gFogBlue, gFogAlpha, gFogNear, gFogFar);
+            }
+            break;
+    }
+    return 0;
+}
+
+extern AnimationHeader D_602AA7C[];
+extern Limb* D_602AB48[];
+
+void func_DF4260_8018F31C(Object_80* obj80) {
+    Vec3f sp28[10];
+
+    Animation_GetFrameData(D_602AA7C, 0, sp28);
```

Comment:
```suggestion
extern AnimationHeader D_602AA7C;
extern Limb* D_602AB48[];

void func_DF4260_8018F31C(Object_80* obj80) {
    Vec3f sp28[10];

    Animation_GetFrameData(&D_602AA7C, 0, sp28);
```
An AnimationHeader is a single struct.

Reply:
oh ok, thought someone mentioned it being an array